hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/realtime/AbstractRealtimeRecordReader.java

```diff
@@ -95,7 +97,7 @@ private boolean usesCustomPayload(HoodieTableMetaClient metaClient) {
 
   private void prepareHiveAvroSerializer() {
     try {
-      List<String> hiveInternalColumns = Arrays.asList(new String[] {"BLOCK__OFFSET__INSIDE__FILE", "INPUT__FILE__NAME", "ROW__ID"});
+      Set<String> hiveInternalColumns = new HashSet(Arrays.asList(new String[] {"BLOCK__OFFSET__INSIDE__FILE", "INPUT__FILE__NAME", "ROW__ID"}));
```

Comment:
I see there are more internal columns https://cwiki.apache.org/confluence/display/hive/languagemanual+virtualcolumns

Also, this could change per Hive version (though not very frequently). How do we plan to maintain it?

Reply:
Thank you very much for your reminder， i think use writeSchema to filter virtualcolumns is more reasonable。 update the code